paper-button.html

```diff
@@ -44,154 +44,58 @@
-      }
-
-      #ripple {
-        pointer-events: none;
-        z-index: -1;
-      }
-
-      .button-content {
-        padding: 0.7em 0.57em
-      }
-
-      polyfill-next-selector { content: '.button-content > a'; }
-      ::content > a {
-        height: 100%;
-        padding: 0.7em 0.57em;
-        margin: -0.7em -0.57em;
-        /* flex */
-        -ms-flex: 1 1 0.000000001px;
-        -webkit-flex: 1;
-        flex: 1;
-        -webkit-flex-basis: 0.000000001px;
-        flex-basis: 0.000000001px;
-      }
-
-    </style>
-
-    <template if="{{raised}}">
-      <paper-shadow id="shadow" fit animated></paper-shadow>
-    </template>
-
-    <!-- this div is needed to position the ripple behind text content -->
-    <div class="button-content" relative layout horizontal center-center>
-      <content></content>
-    </div>
+    <!-- Manage elevation / toggle / raised / aria-pressed -->
+    <paper-button-state
+      elevation="{{elevation}}" toggle="{{toggle}}" raised="{{raised}}"
+      active="[[active]]" focused="[[focused]]" disabled="[[disabled]]">
+      </paper-button-state>
 
-    <core-a11y-keys keys="space enter" target="{{}}" on-keys-pressed="{{_activate}}"></core-a11y-keys>
+    <!-- Enable rippliness -->
+    <paper-ripple></paper-ripple>
 
-  </template>
+    <!-- Reflect elevation -->
+    <paper-card class="content" elevation="[[elevation]]" animated>
 
-  <script>
-    Polymer({
-
-      publish: {
-
-        /**
-         * If true, the button will be styled with a shadow.
-         *
-         * @attribute raised
-         * @type boolean
-         * @default false
-         */
-        raised: false,
-
-        /**
-         * By default the ripple emanates from where the user touched the button.
-         * Set this to true to always center the ripple.
-         *
-         * @attribute recenteringTouch
-         * @type boolean
-         * @default false
-         */
-        recenteringTouch: false,
-
-        /**
-         * By default the ripple expands to fill the button. Set this to true to
-         * constrain the ripple to a circle within the button.
-         *
-         * @attribute fill
-         * @type boolean
-         * @default true
-         */
-        fill: true
-
-      },
-
-      _activate: function() {
-        this.click();
-        this.fire('tap');
-        if (!this.pressed) {
-          var bcr = this.getBoundingClientRect();
-          var x = bcr.left + (bcr.width / 2);
-          var y = bcr.top + (bcr.height / 2);
-          this.downAction({x: x, y: y});
-          var fn = function fn() {
-            this.upAction();
-            this.removeEventListener('keyup', fn);
-          }.bind(this);
-          this.addEventListener('keyup', fn);
-        }
-      }
-
-    });
-  </script>
-</polymer-element>
+      <!-- Re-project light DOM -->
+      <content></content>
+    </paper-card>
+  </template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'paper-button'
```

Comment:
I didn't mean to advocate against 2-way binding all together, only the fact that it gets to *create* properties on the instance that are defined nowhere else.